packages/storybook/stories/components/Textarea.stories.tsx

```diff
@@ -0,0 +1,173 @@
+import tokensDefinition from '@utrecht/components/textarea/tokens.json';
+import tokens from '@utrecht/design-tokens/dist/index.json';
+import React from 'react';
+import { designTokenStory } from './util';
+
+const defaultArgs = {
+  dir: 'auto',
+  disabled: false,
+  focus: false,
+  hover: false,
+  invalid: false,
+  placeholder: undefined,
+  readOnly: false,
+  required: false,
+  spellCheck: false,
+  children: undefined,
+};
+
+const exampleArgs = {
+  children: 'The Quick Brown Fox Jumps Over The Lazy Dog',
+};
+
+const arabicDecorator = (Story) => (
+  <div dir="rtl" lang="ar">
+    {Story()}
+  </div>
+);
+
+const meta = {
+  title: 'CSS Component/Textarea',
+  id: 'css-textarea',
+  component: Textarea,
+  argTypes: {
+    disabled: {
+      description: 'Disabled',
+      control: 'boolean',
+    },
+    invalid: {
+      description: 'Invalid',
+      control: 'boolean',
+    },
+    readOnly: {
+      description: 'Read-only',
+      control: 'boolean',
+    },
+    required: {
+      description: 'Required',
+      control: 'boolean',
+    },
+  },
+  args: {},
+  tags: ['autodocs'],
+  parameters: {
+    status: {
+      type: 'WORK IN PROGRESS',
+    },
+    tokensPrefix: 'utrecht-textarea',
+    tokens,
+    tokensDefinition,
+    docs: {
+      description: {
+        story: `Styling via the \`.utrecht-textarea\` class name.`,
+        component: readme,
+      },
+    },
+  },
+} satisfies Meta<typeof Textarea>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  args: {
+    ...defaultArgs,
+  },
+};
+
+export const DefaultValue: Story = {
+  args: {
+    ...defaultArgs,
+  },
+  render: () => {
+    return <Textarea defaultValue={exampleArgs.children} />;
+  },
+};
+
+export const Direction: Story = {
+  args: {
+    ...defaultArgs,
+  },
+  decorators: [arabicDecorator],
+};
+
+export const Placeholder: Story = {
+  name: 'Placeholder',
+  args: {
+    ...defaultArgs,
+    placeholder: 'Typ je bericht...',
+  },
+};
+
+export const PlaceholderLeftToRight: Story = {
+  name: 'Placeholder (right-to-left)',
+  args: {
+    ...defaultArgs,
+    placeholder: 'اكتب رسالتك',
+  },
+  decorators: [arabicDecorator],
+};
+
+export const Spellcheck: Story = {
+  args: {
+    ...defaultArgs,
+  },
+  argTypes: {
+    ...Default.argTypes,
+    spellCheck: {
+      description: 'Spellcheck',
+      control: { type: 'select' },
+      options: ['', 'false', 'true'],
+    },
+  },
+};
+
+export const Disabled: Story = {
+  args: {
+    ...Default.args,
+    disabled: true,
+  },
+};
+
+export const Invalid: Story = {
+  args: {
+    ...Default.args,
+    invalid: true,
+  },
+};
+
+export const ReadOnly: Story = {
+  args: {
+    ...Default.args,
+    readOnly: true,
+  },
+};
+
+export const Required: Story = {
+  args: {
+    ...Default.args,
+    required: true,
+  },
+};
+
+export const Focus: Story = {
+  args: {
+    ...Default.args,
+    focus: true,
+  },
+};
```

Comment:
`focus` is not an existing property on the React `Textearea` component, this should be part of the CSS story rendering function. The render function should include the `utrecht-textarea--focus` property.

Also note that the current rendering function includes two class names, as you can also spot in the source code for this code snippet.

`focus && 'utrecht-textarea--focus utrecht-textarea--focus-visible',`

Now that we are at it, I think it would be nicer to refactor this into two separate boolean flags, so we can configure them separately for clarity.

```
  args: {
     ...Default.args,
     focus: true,
     focusVisible: true,
   },
```

Now the focus ring is broken 😭

## Main branch

<img width="502" alt="Screenshot 2023-08-29 at 09 21 29" src="https://github.com/nl-design-system/utrecht/assets/30694/e18dbe8b-dc69-4e37-9b57-8d0d6c37083f">

## This branch

<img width="516" alt="Screenshot 2023-08-29 at 09 21 39" src="https://github.com/nl-design-system/utrecht/assets/30694/5be07830-2582-4858-80a9-e71d5c6fcde2">

Reply:
the `showCode` in storybook of the `focus` and `hover` stories have do not show the correct classes by default, since the custom render function seems to clash with `showCode` (it works in the actual preview, just not in the show code). I added a custom dock block in those stories for now. please let me know if you prefer a different approach.